pipelines/ui5-sap-cp/Jenkinsfile

```diff
@@ -13,34 +15,97 @@ import com.sap.piper.Utils
+    return configFile.equals(CONFIG_FILE_YML)
+}
+
 node() {
   //Global variables:
   APP_PATH = 'src'
   SRC = "${WORKSPACE}/${APP_PATH}"
-  CONFIG_FILE = "${SRC}/.pipeline/config.properties"
+
+  CONFIG_FILE_PROPERTIES = '.pipeline/config.properties'
+  CONFIG_FILE_YML = '.pipeline/config.yml'
+
+  def CONFIG_FILE
+
+  def STEP_CONFIG_NEO_DEPLOY='neoDeploy'
+  def STEP_CONFIG_MTA_BUILD='mtaBuild'
 
   stage("Clone sources and setup environment"){
     deleteDir()
+    Map neoDeployConfiguration, mtaBuildConfiguration
     dir(APP_PATH) {
       checkout scm
+      if(fileExists(CONFIG_FILE_YML) ) {
+          CONFIG_FILE = CONFIG_FILE_YML
+      } else if(fileExists (CONFIG_FILE_PROPERTIES) ) {
+          CONFIG_FILE = CONFIG_FILE_PROPERTIES
+      } else {
+          error "No config file found."
+      }
+      echo "[INFO] using configuration file '${CONFIG_FILE}'."
       setupCommonPipelineEnvironment script: this, configFile: CONFIG_FILE
+      prepareDefaultValues script: this
+      neoDeployConfiguration = ConfigurationMerger.merge([:], (Set)[],
+                                                         ConfigurationLoader.stepConfiguration(this, STEP_CONFIG_NEO_DEPLOY), (Set)['neoHome', 'host', 'account', 'neoCredentialsId'],
+                                                         ConfigurationLoader.defaultStepConfiguration(this, 'neoDeploy'))
+      mtaBuildConfiguration = ConfigurationMerger.merge([:], (Set)[],
+                                                        ConfigurationLoader.stepConfiguration(this, STEP_CONFIG_MTA_BUILD), (Set)['mtaJarLocation'],
+                                                        ConfigurationLoader.defaultStepConfiguration(this, 'mtaBuild'))
     }
-    MTA_JAR_LOCATION = commonPipelineEnvironment.getConfigProperty('MTA_HOME')
-    NEO_HOME = commonPipelineEnvironment.getConfigProperty('NEO_HOME')
-    DEPLOY_HOST = commonPipelineEnvironment.getConfigProperty('DEPLOY_HOST')
-    CI_DEPLOY_ACCOUNT = commonPipelineEnvironment.getConfigProperty('CI_DEPLOY_ACCOUNT')
-    neoCredentialsId = commonPipelineEnvironment.getConfigProperty('neoCredentialsId')
+    MTA_JAR_LOCATION = mtaBuildConfiguration.mtaJarLocation ?: commonPipelineEnvironment.getConfigProperty('MTA_HOME')
+    NEO_HOME = neoDeployConfiguration.neoHome ?: commonPipelineEnvironment.getConfigProperty('NEO_HOME')
+    DEPLOY_HOST = neoDeployConfiguration.host ?: commonPipelineEnvironment.getConfigProperty('DEPLOY_HOST')
+    CI_DEPLOY_ACCOUNT = neoDeployConfiguration.account ?: commonPipelineEnvironment.getConfigProperty('CI_DEPLOY_ACCOUNT')
+    neoCredentialsId = neoDeployConfiguration.neoCredentialsId ?: commonPipelineEnvironment.getConfigProperty('neoCredentialsId')
     proxy = commonPipelineEnvironment.getConfigProperty('proxy') ?: ''
     httpsProxy = commonPipelineEnvironment.getConfigProperty('httpsProxy') ?: ''
   }
 
   stage("Validate configuration"){
+
+    def newConfig =  isNewConfigFramework(def configFile)
```

Comment:
here would not be, isNewConfigFramework(CONFIG_FILE)

Reply:
fixed, stupid me!